### DISCUSSION TITLE
How to customize each face of cube

### DISCUSSION THREAD

**Discussion by LucaArgentieri** [Tresjs/tres#596] [Q&A] [answered]
Hi!
I'm trying to customize each face of my cube with different colors (easy step, next step with textures), is it possible to achieve it like [here](https://github.com/pmndrs/react-three-fiber/discussions/2300#discussioncomment-5241166) in r3f 🤔 ?

```
   <TresMesh ref="cubeRef">
       <TresBoxGeometry />
       <TresMeshStandardMaterial attach="material" :color="material1" />
       <TresMeshStandardMaterial attach="material-1" :color="material2" />
       <TresMeshStandardMaterial attach="material-2" :color="material2" />
       <TresMeshStandardMaterial attach="material-3" :color="material2" />
       <TresMeshStandardMaterial attach="material-4" :color="material2" />
       <TresMeshStandardMaterial attach="material-5" :color="material2" />
   </TresMesh>
```

**Comment by andretchen0** [comment, marked as answer]
Hey @LucaArgentieri 

The current version of Tres – to my knowledge – doesn't support pierced props for arrays, e.g., `attach="material-{arrayIndex}"`.

---

@alvarosabu Is this something to add to the core?

## Context

Here's how [R3F handles e.g., "prop-0".](https://github.com/pmndrs/react-three-fiber/blob/e1acae8e1aea6ac8b255d42ffd19d0bfdd6cd529/packages/fiber/src/core/utils.ts#L228)
